lib/msal-core/src/ServerRequestParameters.ts

```diff
@@ -294,11 +245,16 @@ export class ServerRequestParameters {
      * Utility to generate a QueryParameterString from a Key-Value mapping of extraQueryParameters passed
      * @param extraQueryParameters
      */
-    static generateQueryParametersString(queryParameters: StringDict): string {
+    static generateQueryParametersString(queryParameters: StringDict, silentCall?: boolean): string {
         let paramsString: string = null;
 
         if (queryParameters) {
             Object.keys(queryParameters).forEach((key: string) => {
+                // sid cannot be passed along with login_hint or domain_hint
+                if(key == "domain_hint" && (silentCall || paramsString.match("sid"))) {
```

Comment:
Is there an existing constant we can use here instead of the magic string `"domain_hint"`? Same with `"sid"`

Reply:
Also, should use ===

Reply:
Yes. We have constants and we can use them. These changes are first cut.